integration_tests/src/main/python/data_gen.py

```diff
@@ -193,6 +200,8 @@ def with_special_pattern(self, pattern, flags=0, charset=sre_yield.CHARSET, weig
         return self.with_special_case(lambda rand : strs[rand.randint(0, length-1)], weight=weight)
 
     def start(self, rand):
+        if self._is_error:
+            raise NotSupportedInSparkVersion("Collation is only supported on Spark 4.0.0+ in python. All tests that use this feature must have a skipif on them...")
```

Comment:
NIT:
Shall we move this check into the contructor of the StringType (aka in `__init__` method) to fail the app earlier, and the member `_is_error` is no longer needed.